llama/llama.go

```diff
@@ -692,35 +674,52 @@ func SchemaToGrammar(schema []byte) []byte {
 	return buf[:n]
 }
 
-type Sampler struct {
-	c *C.struct_llama_sampler
+type TokenData struct {
+	ID    int32
+	Logit float32
+}
+
+type Grammar struct {
+	c *C.struct_llama_grammar
 }
 
-func NewGrammarSampler(vocab *Vocab, grammar string) *Sampler {
+func NewGrammar(grammar string, vocabIds []uint32, vocabValues []string, eogTokens []uint32) *Grammar {
```

Comment:
```suggestion
func NewGrammar(grammar string, vocabIds []uint32, vocabValues []string, eogTokens []uint32) *Grammar {
```

Why don't we pass a `map[uint32]string` here?